File: tutorial/cpp/stl/containers/set/set.cpp
// SPDX-License-Identifier: BSD-3-Clause

#include <iostream>     // cout
#include <set>          // set
using namespace std;

int main() {
    set<int> s;

    // insert element in set
    for (int i = 0; i < 5; ++i) {
        s.insert(i);
    }

    // erase one element from set
    s.erase(4);

    // add another element to see that elements are sorted
    s.insert(-2);

    // iterate a set
    for (const auto& elem : s) {
        cout << elem << " ";
    }
    cout << "\n";

    return 0;
}

File: tutorial/python/stdlib/containers/set/set.py
#!/usr/bin/env python3
# SPDX-License-Identifier: BSD-3-Clause


def main() -> None:
    s: set[int] = set()
    s.add(1)
    s.add(12)
    s.add(6)
    s.add(1)
    for x in sorted(s):
        print(x)
    # print: 1 6 12


if __name__ == "__main__":
    main()
